tasks/build/packages.js

```diff
@@ -0,0 +1,57 @@
+import { buildProductionProjects } from '@elastic/kbn-build';
+
+/**
+ * High-level overview of how we enable packages in production:
+ *
+ * tl;dr We copy the packages directly into Kibana's `node_modules` folder,
+ * which means they will be available when `require(...)`d.
+ *
+ * During development we rely on `@elastic/kbn-build` to find all the packages
+ * in the Kibana repo and run Yarn in all the right places to create symlinks
+ * between these packages. This development setup is described in-depth in the
+ * readme in `@elastic/kbn-build`.
+ *
+ * However, for production we can't use `@elastic/kbn-build` as part of the
+ * installation as we don't have an install "tool/step" that can kick it off.
+ * We also can't include symlinks in the archives for the different platform, so
+ * we can't run it in the same way we do for development and just package the
+ * result. That means we have two options: either we prepare everything in the
+ * built package or we perform the necessary actions when Kibana is starting up
+ * in production. We decided on the former: all the Kibana packages are
+ * prepared as part of the build process.
+ *
+ * (All of this is a bit different for Kibana plugins as they _do_ have an
+ * install step — the plugin CLI tool. Because of that we can treat them
+ * differently from Kibana itself.)
+ * TODO: Link to the readme that details how we handle plugins.
```

Comment:
Can we link here now? https://github.com/kjbekkelund/kibana/blob/e430aedc1c19bcdfbd7a8a3378a00f241fe233a2/packages/kbn-build/README.md#kibana-plugins

Reply:
++ I plan on going through all the docs and making sure they are both clear enough and link to the right places.